gc/base/HeapVirtualMemory.cpp

```diff
@@ -312,7 +312,19 @@ MM_HeapVirtualMemory::commitMemory(void* address, uintptr_t size)
 {
 	MM_GCExtensionsBase* extensions = MM_GCExtensionsBase::getExtensions(_omrVM);
 	MM_MemoryManager* memoryManager = extensions->memoryManager;
-	return memoryManager->commitMemory(&_vmemHandle, address, size);
+
+	bool resultCommitMemory = memoryManager->commitMemory(&_vmemHandle, address, size);
+
+	if (resultCommitMemory && extensions->pretouchHeapOnExpand) {
+		uintptr_t pageSize = _vmemHandle.getVirtualMemory()->getPageSize();
+		uintptr_t offset = 0;
+		while (offset < size) {
+			memset(static_cast<char*>(address) + offset, 0, 1);
+			offset += pageSize;
+		}
```

Comment:
You can organize this loop better to avoid extra addition inside loop:

    uint8_t *currentAddress = address;
    uint8_t *endAddress = currentAddress + size;
    while (currentAddress < endAddress) {
        memset(currentAddress, 0, 1);
        currentAddress += pageSize;
    }

Reply:
Thinking about this I agree with Aleks that straight `memset(address, 0, size);` is a safe choice. Current implementation with loop has hidden catch on 32-bit platforms: of heap is happen to be located at the very end of virtual memory space it is possible that when `endAddress` is valid `currentAddress += pageSize;` would cause overflow and code will corrupt low addresses. So to do it correct an extra logic required

Reply:
... and to be honest it is my "improved" version of the code suffered from this problem. An original suggested code is free from it. I apologize.

Reply:
@dmitripivkine No worries. I did learn a lot (coding format, efficient logic, better understand of code base).